internal/gatewayapi/route.go

```diff
@@ -516,7 +516,8 @@ func (t *Translator) processGRPCRouteRules(grpcRoute *GRPCRouteContext, parentRe
 
 		// If the route has no valid backends then just use a direct response and don't fuss with weighted responses
 		for _, ruleRoute := range ruleRoutes {
-			if ruleRoute.Destination == nil && ruleRoute.Redirect == nil {
+			noValidBackends := ruleRoute.Destination == nil || ruleRoute.Destination.ToBackendWeights().Valid == 0
```

Comment:
shouldnt this be `invalidBackendsExist` instead of `noValidBackends`  i.e. some instead of all ?

Reply:
noValidBackends = invalidBackendsExist | emptyBackends

Reply:
but `noValidBackends` also implies validBackends == 0 which is not true here

Reply:
in current code validBackends == 0 equals emptyBackends